src/completions.jl

```diff
@@ -76,6 +80,8 @@ function basecompletionadapter(
     CompletionSuggetion[]
   end
 
+  length(METHOD_COMPLETION_CACHE) ≥ MAX_METHOD_COMPLETION_CACHE && empty!(METHOD_COMPLETION_CACHE)
```

Comment:
Could we make this a FIFO cache instead?

Reply:
FIFO needs the cache to be `Vector` instead of `Dict`, doesn't it ?
Or we can hold additional information that represents when a items is `push`ed in the `Dict` and implement FIFO.

Reply:
We can make it an `OrderedDict` (from OrderedCollections.jl), no?

Reply:
Ah yeah, definitely. I didn't know we are using that haha